### ISSUE TITLE
Precondition failed error

### ISSUE THREAD

**Issue by tomhughes** [tomhughes/trac-tickets#1865]
**[Submitted to the original trac issue database at 11.10am, Tuesday, 26th May 2009]**

I got the following error while using Potlatch 1.0:

An unusual error happened (in 'putway' -61). The server said: Precondition failed: Way  requires the nodes with id in (-265), which either do not exist, or are not visible.

Please e-mail richard@systemeD.net with a bug report, saying what you were doing at the time.

I was adding a footway between the end of one road and the middle of another (involved adding a node to the middle.) I got the dreaded red triangle, repeatedly retried. Didn't work. For some reason I decided to delete the footway and try again, reusing the new node in the middle of the road. Error appeared.

Again, this is probably caused by a combination of server load and me doing something rather odd, but we really need to get the upload problems sorted out, which might involve changes to the API??

**Comment by tomhughes** [comment]
**Author: Richard**
**[Added to the original trac issue at 11.15am, Tuesday, 26th May 2009]**

This is a particularly elusive bug which _appears_ to be related to an edge case in splitting ways, but I haven't been able to find steps to reproduce yet. If you could do so that'd be fantastic.

Until then I'm closing it because (as the error message says) I'm trying to keep this one tracked by e-mail, not here.

**Comment by tomhughes** [comment]
**Author: Vid the Kid**
**[Added to the original trac issue at 9.04am, Saturday, 25th July 2009]**

I had a similar issue recently.  I had drawn several new streets in edit-with-save mode, at a time when apparently the server was temporarily down.  Potlatch repeatedly reported failure to load new data, and wouldn't quit trying even when I selected 'cancel'.  When the server came back, every time I tried to upload, I got this error, which sometimes listed one node, sometimes several.  (I had panned away from these streets between drawing them and attempting to upload them.  That might be relevant.)  Potlatch seemed to be refusing to upload the new nodes required by the new ways.  I think server unavailability may be a significant component to this bug.
